# Q&A Activity Question Week -09
Do the following on the dataset
1. Explain the dataset
2. Check the summary statistics and discuss the max, min, avg, median,
and percentiles.
3. The manager wants to find out whether the same amount was spent
for the three advertisements (TV , Radio and Newspaper). Comment
on your findings.

In [1]:
#Importing the necessary libraries 
import pandas as pd
import scipy.stats as stats

In [2]:
#Importing the required dataset
data = pd.read_csv("F:\\pythonprogramming\\Advertising.csv")
data.head()

Unnamed: 0     TV  Radio  Newspaper  Sales
0           1  230.1   37.8       69.2   22.1
1           2   44.5   39.3       45.1   10.4
2           3   17.2   45.9       69.3    9.3
3           4  151.5   41.3       58.5   18.5
4           5  180.8   10.8       58.4   12.9

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  200 non-null    int64  
 1   TV          200 non-null    float64
 2   Radio       200 non-null    float64
 3   Newspaper   200 non-null    float64
 4   Sales       200 non-null    float64
dtypes: float64(4), int64(1)
memory usage: 7.9 KB


# 1.  Explain the dataset

In [4]:
#Printing the dataset
data

Unnamed: 0     TV  Radio  Newspaper  Sales
0             1  230.1   37.8       69.2   22.1
1             2   44.5   39.3       45.1   10.4
2             3   17.2   45.9       69.3    9.3
3             4  151.5   41.3       58.5   18.5
4             5  180.8   10.8       58.4   12.9
..          ...    ...    ...        ...    ...
195         196   38.2    3.7       13.8    7.6
196         197   94.2    4.9        8.1    9.7
197         198  177.0    9.3        6.4   12.8
198         199  283.6   42.0       66.2   25.5
199         200  232.1    8.6        8.7   13.4

[200 rows x 5 columns]

In [5]:
# To verify null data items
data.isnull().sum() 

Unnamed: 0    0
TV            0
Radio         0
Newspaper     0
Sales         0
dtype: int64

The given dataset consists of advertisement expenses on three different medias(TV,Radio and Newspaper) and the sales.

There are 200 rows and 5 columns of data and there are no null values in this dataset

# 2.  Check the summary statistics and discuss the max, min, avg, median, and percentiles.

In [6]:
data.describe()

Unnamed: 0          TV       Radio   Newspaper       Sales
count  200.000000  200.000000  200.000000  200.000000  200.000000
mean   100.500000  147.042500   23.264000   30.554000   14.022500
std     57.879185   85.854236   14.846809   21.778621    5.217457
min      1.000000    0.700000    0.000000    0.300000    1.600000
25%     50.750000   74.375000    9.975000   12.750000   10.375000
50%    100.500000  149.750000   22.900000   25.750000   12.900000
75%    150.250000  218.825000   36.525000   45.100000   17.400000
max    200.000000  296.400000   49.600000  114.000000   27.000000

The summary statistics for the given dataset:
    
The dataset consists of 200 rows and 5 columns. Maximum amount spent on advertisement was on TV and minimum was spent on radio.

Overall mean is 100.50 , TV  147.04, Radio 23.26, and Newspaper 30.55.

The median is TV 149.75 , Radio 22.9  and  Newspaper 25.75.

75% of the expense on TV lies below 218.825 , Radio 36.525 and Newspaper 45.10

# 3. The manager wants to find out whether the same amount was spent for the three advertisements (TV , Radio and Newspaper). Comment on your findings.

# One way ANOVA 

NUll Hypothesis:   Same amounts are spent on 3 advertisements
    
Alternate Hypothesis:     Different amounts were spent on 3 advertisements 


In [14]:
stats.f_oneway(data['TV'],data['Radio'],data['Newspaper'],axis=0)

F_onewayResult(statistic=358.8514595342597, pvalue=4.552931539744962e-103)

 pvalue  =4.552931539744962e-103 which is < 0.05 , so we reject the null hypothesis and retain the alternate hypothesis ie, different amount is spent on 3 ads.
    ( F value is inversely related to p value and higher F value (greater than F critical value) indicates a significant p value.)